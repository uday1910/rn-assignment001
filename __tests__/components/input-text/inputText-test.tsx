import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import InputText from '../../../source/components/input-text/inputText';

const inputStyle = {
  color: 'white',
  padding: 20,
};

const mockTextChangeEvent = jest.fn();

const props = {
  placeholder: 'Enter the value',
  style: inputStyle,
  maxLength: 20,
  onChangeText: mockTextChangeEvent,
};

describe('InputText Function', () => {
  it('InputText render properly according to the snapshot', () => {
    const inputText = renderer.create(<InputText {...props} />).toJSON();
    expect(inputText).toMatchSnapshot();
  });

  it('On change text event called while typing', () => {
    const {getByPlaceholderText} = render(<InputText {...props} />);
    const textInputObj = getByPlaceholderText('Enter the value');
    fireEvent(textInputObj, 'onChangeText');
    expect(mockTextChangeEvent).toHaveBeenCalled();
  });

  it('Trigger text change event in Text input', () => {
    const {getByTestId} = render(<InputText {...props} />);
    const textInput = getByTestId('input-text');
    expect(textInput).toBeTruthy();
    fireEvent.changeText(textInput, 'uday');
  });
});
