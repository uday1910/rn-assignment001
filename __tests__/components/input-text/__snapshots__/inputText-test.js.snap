// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText Function InputText render properly according to the snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  autoCorrect={false}
  maxLength={20}
  onChangeText={[Function]}
  placeholder="Enter the value"
  placeholderTextColor="#a0a0a0"
  rejectResponderTermination={true}
  returnKeyLabel="Done"
  returnKeyType="done"
  style={
    Object {
      "borderBottomColor": "#a0a0a0",
      "borderBottomWidth": 0.5,
      "color": "white",
      "fontSize": 18,
      "marginBottom": 10,
      "padding": 20,
      "width": "85%",
    }
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
